### PR TITLE
prevent overflow in blip buffer size calculation

### PR DESCRIPTION
**Pull request by jmestwa-coder**
### Summary
This change updates buffer size calculations in HSLF blip processing to avoid potential integer overflow before array allocation.

### What changed

\- Replaced inline size calculations using \`new byte\[...\]\` with:

\- \`long\`-based size computation

\- \`IOUtils.safelyAllocate(...)\` for allocation

\- Applied consistently to:

\- \`Bitmap\`

\- \`EMF\`

\- \`PICT\`

\- \`WMF\`

### Why

In the current implementation, buffer sizes are computed using integer arithmetic:

new byte\[a + b + c\]

If intermediate values exceed Integer.MAX\_VALUE, this can overflow before allocation, leading to incorrect buffer sizes and runtime failures.

Using IOUtils.safelyAllocate ensures:

*   size is evaluated using long
    
*   overflow is detected before allocation
    
*   behavior is consistent with existing POI safety patterns
    

### Scope

*   Limited to HSLF blip classes only
    
*   No API changes
    
*   No behavioral changes for valid inputs
    

### Notes

This follows the same allocation safety approach already used in other parts of POI.

### CODE REVIEW

File: poi-scratchpad/src/main/java/org/apache/poi/hslf/blip/WMF.java [hunk 1]
```diff
@@ -90,7 +90,8 @@ protected byte[] formatImageForSlideshow(byte[] data) {
         header.setZipSize(compressed.length);
 
         byte[] checksum = getChecksum(data);
-        byte[] rawData = new byte[checksum.length * getUIDInstanceCount() + header.getSize() + compressed.length];
+        long size = (long)checksum.length * getUIDInstanceCount() + header.getSize() + compressed.length;
```
Comment (pjfanning) [inline poi-scratchpad/src/main/java/org/apache/poi/hslf/blip/WMF.java]: Move this calculation and the other new calculations in the PR to a shared method that can be tested in your tests. Your tests don't seem to test the code and don't seem to provide any regression protection for the future. I only have my phone with me so I may have missed something but my cursory check of the new code makes me suspicious that this new code is maintainable.

Comment (jmestwa-coder) [reply poi-scratchpad/src/main/java/org/apache/poi/hslf/blip/WMF.java]: I updated the calculations to use Math.addExact for clearer, overflow-safe logic and kept the change minimal without adding a new helper. I also updated the test to cover the actual calculation path and verify the overflow behaviour.

**Comment by pjfanning** [comment]
I went with https://github.com/apache/poi/pull/1054 instead.
